lockfile/src/cargo.rs

```diff
@@ -0,0 +1,164 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+use serde::Deserialize;
+
+use crate::{Parse, ParseResult};
+
+pub struct Cargo;
+
+#[derive(Deserialize, Debug, Clone)]
+struct CargoLock {
+    #[serde(rename = "package")]
+    packages: Vec<Package>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct Package {
+    name: String,
+    version: String,
+    source: Option<String>,
+}
```

Comment:
Could we rely on Cargo's internal structs rather than making our own? Like https://docs.rs/cargo/0.65.0/cargo/core/resolver/struct.EncodableResolve.html?

Reply:
I don't see how to use EncodableResolve, the fields are there but they're all private. If you have suggestions I'm up for implementing them.

Reply:
You could deserialize and convert it to a `Resolve`. But it might be easier to have our own version. I was just curious if you've looked into this at all.

Reply:
We looked into it, and it seems like more trouble than it would be worth. We would need to update the parse() function to include a path to the cargo.toml file that we want to parse, whereas we're currently passing in the contents of the lockfile as a string. We would also need to link against the Cargo crate, which we don't do yet.